.github/workflows/tkg_integration_tests_clusterclass.yaml

```
@@ -40,6 +40,120 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  integ-tests-docker-cc:
+    name: TKG Integration Tests CAPD (ClusterClass)
+    runs-on: tkg
+    steps:
+      - name: Free disk space and clean up installed plugins on runner
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /home/tanzu/.cache
+          sudo rm -rf /home/tanzu/.kube
+          sudo rm -rf /home/tanzu/.local/share/tanzu-cli/*
+          sudo sh -c 'find /tmp -type d -name "go-build*" -mtime +3 | xargs rm -rf'
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.17'
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+
+      - name: go cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Get dependencies
+        run: |
+          go get -v -t -d ./...
+          if [ -f Gopkg.toml ]; then
+              curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+              dep ensure
+          fi
+
+      - name: Build Tools
+        run: make tools
+
+      - name: Extract PR Number
+        uses: Dovyski/payload-info-action@master
+        id: get_pr_num
+        with:
+          # when event is a pull request, obtaining the PR number is obvious
+          filter_pull_request: '.number'
+          # when event is a push (merge of PR), since we require linear history,
+          # we are not even generating a merge commit that can help identify the
+          # PR number, so don't even try. Instead we just hard-code to a dummy
+          # value.
+          filter_push: 1009
+
+      - name: Find Comment
+        uses: peter-evans/find-comment@v1
+        id: findcomment
+        with:
+          issue-number: ${{ steps.get_pr_num.outputs.value }}
+          body-includes: //usebom
+          direction: last
+
+      - name: Extract Alternate BOM
+        shell: bash
+        run: |
+          export BOMCOMMENT=$(echo -e "${{ steps.findcomment.outputs.comment-body }}" | tr -d "[:space:]")
+          echo "##[set-output name=bompath;]$(echo "$BOMCOMMENT" | awk -F : '{print $2}')"
+        id: extract_bom
+
+      - name: Login to GCR
+        if: ${{ env.gcp_bucket_sa != '' }}
+        uses: docker/login-action@v2
+        with:
+          registry: gcr.io
+          username: _json_key
+          password: ${{ secrets.GCP_BUCKET_SA }}
+
+      - name: Build management packages
+        env:
+          NUM: ${{ github.run_id }}
+          OCI_REGISTRY: ${{ steps.set_registry.outputs.registry }}/tanzu_framework/github-actions/${{ steps.extract_reg_dir.outputs.regdir }}
```

We should use different path for CAPD and CAPA package building. May be something like appending `capd` for capd tests?

```suggestion
          OCI_REGISTRY: ${{ steps.set_registry.outputs.registry }}/tanzu_framework/github-actions/${{ steps.extract_reg_dir.outputs.regdir }}/capd
```

And for existing CAPA tests, we can append `capa` to the registry path so that both pipeline doesn't use same path to publish packages? cc @vuil

---

thinking out loud here:
we are talking about the same workflow file here right? 
why would we need to build and publish twice?

---

This kinda makes sense to me. Will wait for @vuil 's input on adding this

---

My input is not to build publish twice so we don't even have this problem to begin with.
Factor out the part where we build/publish the management repo, then have other jobs depend on it.
Check out https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds